wiki/Ranking_criteria/osu!mania/en.md

```diff
@@ -58,13 +63,14 @@ Overall rules and guidelines apply to every kind of osu!mania difficulty. Rhythm
 - **Two notes cannot be placed in a single column at the same timestamp.**
 - **No column can be left empty.** If you want to use fewer keys, change the number of keys in the Song Setup.
 - **There must not be more than 6 notes pressed at any given moment in Insane or lower difficulties.** Using more than 6 notes must also follow a reasonable spread to the next lower difficulty. This rule does not apply to ends of long notes, as they are released, not pressed.
-- **Beatmaps can only use 4 to 10 keys.** Anything else is not supported for the Ranked section.
-- **If multiple key modes are present in a single beatmap, the key mode must be denoted in all difficulty names. Otherwise, the key mode must not be denoted.**
-- **Beatmaps using the special N+1 style of mapping must be denoted as such under the `Advanced` tab of song setup in the editor and "N+1" must be added to the tags. For beatmaps containing multiple key modes, difficulties using this style must be denoted using N+1 notation (e.g. using 7K+1 instead of 8K on difficulty names).**
+- **Beatmaps can only use 4–10, 12, 14, 16, or 18 keys.**
+- **Each difficulty's name must denote its [playstyle](#common-terms), but only if its key count is above 10 or multiple playstyles are present in the beatmap.**
+- **Each non-[standard playstyle](#common-terms) present in a beatmap must be denoted in the beatmap's tags.**
+- **Difficulties using an [N+1 playstyle](#common-terms) must enable the `Use special style (N+1 style) for mania` option under the `Advanced` tab of the `Song Setup` window. The beatmap must also include the "N+1" tag.** For the purposes of this rule, only playstyles with key counts below 10 can be considered "N+1".[^n-plus-1]
```

Comment:
`Difficulties` should probably be changed to `Beatmaps`. It's making reference to individual simfiles, but it's somewhat more correct to refer to them by Beatmaps rather than Difficulties to not mix the meaning of Difficulty

```suggestion
- **Beatmaps using an [N+1 playstyle](#common-terms) must enable the `Use special style (N+1 style) for mania` option under the `Advanced` tab of the `Song Setup` window. The beatmap must also include the "N+1" tag.** For the purposes of this rule, only playstyles with key counts below 10 can be considered "N+1".[^n-plus-1]
```

Reply:
the setting mentioned in the rule is per individual difficulty

Reply:
It should probably be used for Beatmaps as a whole, specially now with the wording changes that would consider a "N+1" variation of a key mode its own playstyle. You don't really apply this arbitrarily on a per-difficulty basis. Maybe I'm looking too into it though.

Reply:
You can have both 7K+1 and 8K in one mapset, but you only apply special style tick to 7K+1 beatmaps. Applying this to the *whole* mapset defeats the purpose of special style because some mapsets that include N+1 may have a variant that is on different keycount, e.g 4K and 5K+1 mapset.

Reply:
Hmm always had it wrong then, thinking that a Beatmap != Beatmapset when both meant the same thing. Then this is correct yep.